ddtrace/llmobs/_integrations/utils.py

```diff
@@ -397,6 +397,209 @@ def openai_set_meta_tags_from_chat(span: Span, kwargs: Dict[str, Any], messages:
+                pass
+
+        if output.type in ["function_call", "file_search_call", "web_search_call", "computer_call"]:
+            tool_call_info = []
+            output_type = output.type
+            # file_search_call
+            file_queries_raw = _get_attr(output, "queries", None)
+            # file_queries is an array of strings
+            if file_queries_raw:
+                file_queries = {"queries": file_queries_raw}
+
+            # computer_call action
+            action_raw = _get_attr(output, "action", "")
+            if action_raw:
+                if isinstance(action_raw, str):
+                    action_json_str = action_raw.json()
+                    # convert to valid json string
+                    action = json.loads(f'{{"action": {action_json_str}}}')
+
+            # function call
+            # Use output.id as the tool_id
+            tool_id = _get_attr(output, "id", "")
+            tool_name = _get_attr(output, "name", "")
+            arguments = _get_attr(output, "arguments", "")
+            if _get_attr(output, "arguments", ""):
+                arguments = json.loads(arguments)
+            # Set reasoning action as the parameter
+            elif _get_attr(output, "action", ""):
+                arguments = action
+            # Set file search queries as the parameter
+            elif _get_attr(output, "queries", None):
+                arguments = file_queries
+
+            tool_call = {
+                "name": tool_name,
+                "tool_id": tool_id,
+                "type": output_type,
+            }
+            if arguments:
+                tool_call["arguments"] = arguments
+            tool_call_info.append(tool_call)
+            arguments = tool_call.get("arguments", {})
+            # Convert arguments to string before dispatching
+            arguments_str = json.dumps(arguments) if isinstance(arguments, (dict, list)) else str(arguments)
+            core.dispatch(
+                DISPATCH_ON_LLM_TOOL_CHOICE,
+                (
+                    tool_id,
+                    tool_name,
+                    arguments_str,
+                    {
+                        "trace_id": format_trace_id(span.trace_id),
+                        "span_id": str(span.span_id),
+                    },
+                ),
+            )
+            if tool_call_info:
+                output_messages.append({"content": "", "tool_calls": tool_call_info})
+                core.dispatch(DISPATCH_ON_TOOL_CALL_OUTPUT_USED, (tool_id, span))
+        span._set_ctx_item(OUTPUT_MESSAGES, output_messages)
+
+
+def openai_construct_response_from_streamed_chunks(streamed_chunks: List[Any]) -> Dict[str, Any]:
+    """Constructs a response dictionary from streamed chunks.
+    The resulting response dictionary is of form:
+    {"output": [{"role": "...", "content": [{"text": "..."}]}]}
+    """
+    message: Dict[str, Any] = {"content": "", "tool_calls": []}
+
+    for chunk in streamed_chunks:
```

Comment:
Like I mentioned above it seems like we can just rely on the very last `response.completed` chunk which already has the complete response object (i.e. we don't need to stitch any response chunks together like we have to do with completions/chat chunks). Given this, do we still need to iterate through streamed_chunks?